sdk/samples/iot/iot_sample_common.c

```diff
@@ -82,7 +82,7 @@ static az_result read_configuration_entry(
     (void)printf("%s = %s\n", env_name, hide_value ? "***" : env_value);
     az_span env_span = az_span_create_from_str(env_value);
 
-    AZ_RETURN_IF_NOT_ENOUGH_SIZE(destination, az_span_size(env_span));
+    IOT_SAMPLE_RETURN_IF_NOT_ENOUGH_SIZE(destination, az_span_size(env_span));

```

Comment:
Make the names friendlier.  Remove the `IOT_SAMPLE` portion of the macros `RETURN_IF_NOT_ENOUGH_SIZE`.  If customers copy the code and use it they may also want the MACROS in the sample.

Reply:
@momuno - you came up with this naming prefix recently. Should I keep `IOT_SAMPLE`, or make it something else?

Reply:
I contacted Mollie, we will likely follow up outside of this PR.

Reply:
I suggest the naming as when code is copied it makes sense for the user w/o needing to rename/replace the MACRO name. 
 IoT can make the final decision on that.

Reply:
I chose to go with IOT_SAMPLE on macros to follow the iot_sample prefix used on everything else in the iot_sample_commong.c and .h files.  There are many places that AZ_ prefixed functions are called in the sample, such as with json parsing.  Why does this MACRO name need to change?

Reply:
> Why does this MACRO name need to change?

We made the `AZ_RETURN_IF_NOT_ENOUGH_SIZE` macro internal. So, similar to `IOT_SAMPLE_PRECONDITION_NOT_NULL`, the samples need to define their own macro.